### PR TITLE
Bug/add deny modifier in transfer method

### PR DESCRIPTION
**Pull request by chiro-hiro**

### CODE REVIEW

**Comment by chiro-hiro** [comment]
I've changed the order of parameters in `hasPermission()` from `hasPermission(actor, owner, required)` to `hasPermission(owner, actor, required)` to keep it consistent with `delegated(owner, actor)`.

File: contracts/EIP-6366/EIP-6366-Core.sol [hunk 1]
```diff
@@ -1,12 +1,12 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.7;
-import './interfaces/IERC6366Error.sol';
-import './interfaces/IERC6366Core.sol';
+import './interfaces/IEIP-6366-Error.sol';
```
Comment (vdusart) [inline contracts/EIP-6366/EIP-6366-Core.sol]: I think it's weird to add a `-` in the file name and not in the contract name
We should keep the name like `IEIP6366Error.sol`

File: contracts/EIP-6366/EIP-6366-Meta.sol [hunk 2]
```diff
@@ -1,12 +1,12 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.7;
-import './interfaces/IERC6366Meta.sol';
-import './interfaces/IERC6366Error.sol';
+import './interfaces/IEIP-6366-Meta.sol';
```
Comment (vdusart) [inline contracts/EIP-6366/EIP-6366-Meta.sol]: same as before

File: contracts/example/APermissioned.sol [hunk 3]
```diff
@@ -1,13 +1,13 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.7;
-import '../ERC6366/interfaces/IERC6366Core.sol';
-import '../ERC6366/interfaces/IERC6366Error.sol';
+import '../EIP-6366/interfaces/IEIP-6366-Core.sol';
```
Comment (vdusart) [inline contracts/example/APermissioned.sol]: same

File: contracts/example/APermissionToken.sol [hunk 4]
```diff
@@ -1,12 +1,18 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.7;
-import '../ERC6366/ERC6366Core.sol';
-import '../ERC6366/ERC6366Meta.sol';
+import '../EIP-6366/EIP-6366-Core.sol';
+import '../EIP-6366/EIP-6366-Meta.sol';
```
Comment (vdusart) [inline contracts/example/APermissionToken.sol]: same

File: contracts/example/AEcosystem.sol [hunk 5]
```diff
@@ -7,23 +7,23 @@ contract AEcosystem is APermissioned {
     // Constructor code
   }
 
-  function createProposal(address _permissionOwner) external allow(_permissionOwner, PERMISSION_CREATE) {
+  function createProposal(address _permissionOwner) external allow(_permissionOwner, PERMISSION_CREATE) notBlacklisted {
```
Comment (vdusart) [inline contracts/example/AEcosystem.sol]: I would have reversed the two to first check if the person is blacklist before applying the other modifier

Comment (vdusart) [reply contracts/example/AEcosystem.sol]: the two modifers*
```solidity
function createProposal(address _permissionOwner) external notBlacklisted allow(_permissionOwner, PERMISSION_CREATE)  {
```

File: contracts/example/APermissionToken.sol [hunk 6]
```diff
@@ -76,7 +92,7 @@ contract APermissionToken is ERC6366Core, ERC6366Meta {
   function transfer(
     address _to,
     uint256 _permission
-  ) external override allow(PERMISSION_TRANSFER) returns (bool result) {
+  ) external override allow(PERMISSION_TRANSFER) notBlacklisted returns (bool result) {
```
Comment (vdusart) [inline contracts/example/APermissionToken.sol:95]: here swap them too

File: contracts/EIP-6366/EIP-6366-Core.sol (hunk 1 above)
Comment (chiro-hiro) [reply contracts/EIP-6366/EIP-6366-Core.sol]: I see the CI/CD of ethereum/EIPs was failed due to the checking for related link.
https://github.com/ethereum/EIPs/actions/runs/4092373742/jobs/7057063640

File: contracts/EIP-6366/EIP-6366-Meta.sol (hunk 2 above)
Comment (chiro-hiro) [reply contracts/EIP-6366/EIP-6366-Meta.sol]: Got it, let me check the CI/CD report

File: contracts/example/AEcosystem.sol (hunk 5 above)
Comment (chiro-hiro) [reply contracts/example/AEcosystem.sol]: I will update

File: contracts/example/APermissionToken.sol (hunk 6 above)
Comment (chiro-hiro) [reply contracts/example/APermissionToken.sol:95]: Got it

**Review by vdusart — APPROVED** [review]

**Pull request merged**